src/formatting_stack/strategies/impl.clj

```diff
@@ -38,13 +48,12 @@
                                              true
                                              (-> s File. .exists)))))
 
-(speced/defn ^::existing-files absolutize [command, ^::file-entries file-entries]
-  (let [toplevel-fragments (case command
-                             "git" (-> (sh "git" "rev-parse" "--show-toplevel")
-                                       (:out)
-                                       (string/split #"\n")
-                                       (first)
-                                       (string/split separator-pattern)))]
+(speced/defn ^::existing-files absolutize [^::file-entries file-entries]
```

Comment:
`strategies/impl` is git-agnostic (`"git"` is only ever handled over a `case`), while `stategies.clj` makes git part of its explicit api (e.g. a given defn is called `git-completely-staged`, not `completely-staged`).

So as of `master`, we're very clear about what we promise / intend to offer: an opt-in git interface, and some agnostic implementation details.

I wouldn't deviate from that as f-s, should be usable in non-git environments (e.g. some enterprises).

---

(speaking of, we can make that `case` a multimethod instead - only an _open_ set is truly agnostic)

Reply:
As eluded to here: https://github.com/nedap/formatting-stack/pull/185#discussion_r601425939; This change might look like flexibility is removed, but the flexibility is a false promise. The function is not open for extension, and only requires any consumer to provide a static argument (`"git"` in this case). Consumers without git would need their own replacements for the strategies anyway. 

This is an internal function anyway; which is subject to not public by definition. While I agree with the sentiment it's a opt-in git interface, i don't agree that this change affects that in any way.

Reply:
>  but the flexibility is a false promise. 

It trivially becomes a true promise by refactoring a single `case` to a multimethod. We can be thankful for having discovered that via this PR.

> While I agree with the sentiment it's a opt-in git interface, i don't agree that this change affects that in any way.

It does increase the cost of implementing non-git strategies. As of this branch, if tomorrow someone created a PR with `hg` support, we'd plausibly have a hard time adapting internals, tests, etc because everything was more coupled to `git` than before.

Reply:
> It trivially becomes a true promise by refactoring a single case to a multimethod. We can be thankful for having discovered that via this PR.

I don't think it's something we should pursue. The current naming of te strategies are not abstracted properly from git. Mecurial for example, doesn't have any notion of 'staged' files. Refactoring the case to a multimethod does not fix that.

> It does increase the cost of implementing non-git strategies.

I'm not sure about this. I prefer to have small, specific namespaces. If another vcs would be supported i would not put it in `formatting-stack.strategies`, but in a more specific one. Especially if you keep the opt-in promise in mind. 

I don't see how the amount of work would be affected.

Reply:
Ok but let's rename absolutize -> git-absolutize

Also https://github.com/nedap/formatting-stack/pull/184 should be merged before this one, since that one precisely adds coverage for this sort of things

Reply:
This change was mostly driven from the necessity to run the `git` command in a specific directory for both the test-suite and the code. Which in turn, presented the opportunity remove the `"git"`-definition and dependent code. This is extracted into 
 e61c276794575714f3015c489f8b8e241f8b204


I discovered `clojure.java.shell/with-sh-dir`, which perfectly suits our usecase here, so i undid related changes.